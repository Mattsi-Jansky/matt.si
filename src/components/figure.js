import React from 'react'
import { graphql, StaticQuery } from 'gatsby'
import Img from 'gatsby-image'

const InnerFigure = ({ data, children, width }) => {
  if(!width) width = "100%"
  return (
    <div id="content-figure" style={{width: width}}>
      <figure>
        <Img fluid={data} onClick={() => { window.open(data.src)}} />
        {children && <figcaption>{children}</figcaption>}
      </figure>
    </div>)
}

const renderFigureWithQueryResults = (src, children, width) => data => {
  var queryResult = data.figureQuery.edges.find(
    element => element.node.fluid.src.split('/').pop() === src
  )
  if (!queryResult) throw new Error("Figure Comoponent: Can't find image")
  return <InnerFigure children={children} data={queryResult.node.fluid} width={width} />
}

export default ({ children, src, width }) => (
  <StaticQuery
    query={graphql`
      query FigureQuery {
        figureQuery: allImageSharp {
          edges {
            node {
              fluid(maxWidth: 1240) {
                ...GatsbyImageSharpFluid
              }
            }
          }
        }
      }
    `}
    render={renderFigureWithQueryResults(src, children, width)}
  />
)
export { InnerFigure }
