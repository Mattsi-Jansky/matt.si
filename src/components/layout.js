import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import { StaticQuery, graphql } from 'gatsby'
import Helmet from 'react-helmet'

import Sidebar from '.././components/sidebar'
import '../styles/main.scss'
import '../styles/fonts/font-awesome/css/font-awesome.min.css'

const InnerLayout = ({ data, children }) => (
  <Fragment>
    <Helmet>
      <link
        href="https://fonts.googleapis.com/css?family=Lato|PT+Serif&display=swap"
        rel="stylesheet"
      />
    </Helmet>
    <Sidebar siteMetadata={data.site.siteMetadata} />
    <main className="clearfix">{children}</main>
  </Fragment>
)

const DefaultLayout = ({ children }) => (
  <StaticQuery
    query={graphql`
      query SiteTitleQuery {
        site {
          siteMetadata {
            author
            description
            social {
              twitter
              facebook
              linkedin
              github
              email
            }
          }
        }
      }
    `}
    render={data => <InnerLayout children={children} data={data} />}
  />
)

DefaultLayout.propTypes = {
  children: PropTypes.node.isRequired,
}

export default DefaultLayout
export { InnerLayout }
