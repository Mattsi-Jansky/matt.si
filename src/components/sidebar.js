import { Link } from 'gatsby'
import React from 'react'

const Sidebar = ({ siteMetadata }) => (
  <aside className="sidebar">
    <header>
      <div className="about">
        <div className="cover-author-image">
          <Link to="/">
            <img
              src="/pic.jpg"
              alt={siteMetadata.author}
              className="author-image"
            />
          </Link>
        </div>
        <div className="author-name">{siteMetadata.author}</div>
        <p>
          I'm a Software Craftsman at <a href="https://codurance.com/" target="_blank">Codurance</a>. I love solving problems, learning, and empowering my team.
        </p>
      </div>
    </header>
    <footer>
      <section className="contact">
        <hr />
        <ul>
          {siteMetadata.social.twitter && (
            <li>
              <a
                href={`https://twitter.com/${siteMetadata.social.twitter}`}
                target="_blank"
              >
                <i className="fa fa-twitter" aria-hidden="true" />
              </a>
            </li>
          )}
          {siteMetadata.social.facebook && (
            <li>
              <a
                href={`https://facebook.com/${siteMetadata.social.facebook}`}
                target="_blank"
              >
                <i className="fa fa-facebook" aria-hidden="true" />
              </a>
            </li>
          )}
          {siteMetadata.social.github && (
            <li>
              <a
                href={`https://github.com/${siteMetadata.social.github}`}
                target="_blank"
              >
                <i className="fa fa-github" aria-hidden="true" />
              </a>
            </li>
          )}
          {siteMetadata.social.linkedin && (
            <li>
              <a
                href={`https://linkedin.com/in/${siteMetadata.social.linkedin}`}
                target="_blank"
              >
                <i className="fa fa-linkedin" aria-hidden="true" />
              </a>
            </li>
          )}
          {siteMetadata.social.email && (
            <li>
              <a href={`mailto:${siteMetadata.social.email}`} target="_blank">
                <i className="fa fa-envelope-o" aria-hidden="true" />
              </a>
            </li>
          )}
          <li>
            <a href={`/rss.xml`} target="_blank">
              <i className="fa fa-rss" aria-hidden="true" />
            </a>
          </li>
        </ul>
      </section>
      <div className="copyright">
        <p>
          Made with <a href="https://www.gatsbyjs.org">Gatsby</a> and{' '}
          <a href="https://github.com/wangonya/flexible-gatsby/">Flexible</a>
        </p>
      </div>
    </footer>
  </aside>
)

export default Sidebar
