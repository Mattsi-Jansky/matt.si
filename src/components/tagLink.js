import React from 'react'
import { Link } from 'gatsby'
import generatePathToTag from '../shared/pathToTagGenerator'

const TagLink = ({ tag, children }) => (
  <Link to={generatePathToTag(tag)}>{children}</Link>
)

export default TagLink
