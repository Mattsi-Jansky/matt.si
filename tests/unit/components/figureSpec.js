import React from "react"
import renderer from "react-test-renderer"
import { InnerFigure } from "../../../src/components/figure"
import { shallow } from 'enzyme';
import expectExport from "expect";

const testData = {
  "base64":"data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAALABQDASIAAhEBAxEB/8QAFwABAQEBAAAAAAAAAAAAAAAAAwABBP/EABUBAQEAAAAAAAAAAAAAAAAAAAAB/9oADAMBAAIQAxAAAAES6MFko//EABoQAAICAwAAAAAAAAAAAAAAAAECAAMREyH/2gAIAQEAAQUCNx17OclKKy4AmJ//xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAEDAQE/AT//xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAECAQE/AT//xAAeEAACAQMFAAAAAAAAAAAAAAAAARICIYEQEUFRYf/aAAgBAQAGPwKLqt6Rkng5wi6Kdu9P/8QAHRAAAgEEAwAAAAAAAAAAAAAAAAERITFBUWGBkf/aAAgBAQABPyFm4M1E7Sb2Oc+wamtug3YokOW/T//aAAwDAQACAAMAAAAQEz//xAAXEQADAQAAAAAAAAAAAAAAAAAAAREh/9oACAEDAQE/EGrpEf/EABYRAQEBAAAAAAAAAAAAAAAAAAABEf/aAAgBAgEBPxCVr//EAB8QAQACAQMFAAAAAAAAAAAAAAEAESExQVFhcZHR8f/aAAgBAQABPxBeEekldrM/ICQMFAPF2wuXceUeSFWGDpt0imNAa3ISy441+5//2Q==",
  "aspectRatio":1.8283870967741935,
  "src":"/static/45b26cc583de29f73117dce640a86d75/8a760/banner.jpg",
  "srcSet":"/static/45b26cc583de29f73117dce640a86d75/68709/banner.jpg 310w,\n/static/45b26cc583de29f73117dce640a86d75/53593/banner.jpg 620w,\n/static/45b26cc583de29f73117dce640a86d75/8a760/banner.jpg 1240w,\n/static/45b26cc583de29f73117dce640a86d75/0e6ff/banner.jpg 1860w,\n/static/45b26cc583de29f73117dce640a86d75/7efdd/banner.jpg 2480w,\n/static/45b26cc583de29f73117dce640a86d75/f2bea/banner.jpg 2834w",
  "sizes":"(max-width: 1240px) 100vw, 1240px"
}

describe("Figure", () => {
  it("hasn't changed", () => {
    const figure = renderer
      .create(<InnerFigure data={testData}>Image from <a href="foo.bar">Foo Bar</a></InnerFigure>)
      .toJSON()
    expect(figure).toMatchSnapshot()
  })

  it("should include a Gatsby Img with fluid prop", () => {
    const figure = shallow(<InnerFigure data={testData}/>)

    expect(figure.exists('Image')).toEqual(true)
    expect(figure.find('Image').prop('fluid')).toEqual(testData)
  })

  it("should render children inside figcaption", () => {
    const expectedText = "Image from Foo Bar"
    const figure = shallow(<InnerFigure data={testData}>{expectedText}</InnerFigure>)

    expect(figure.exists('figcaption')).toEqual(true)
    expect(figure.find('figcaption').text()).toEqual(expectedText)
  })

  it("given no caption is passed should not render figcaption element", () => {
    const figure = shallow(<InnerFigure data={testData}/>)

    expect(figure.exists('figcaption')).toEqual(false)
  })

  it("given width prop should set width of figure through wrapper", () => {
    const expectedWidth = 700
    const figure = shallow(<InnerFigure data={testData} width={expectedWidth}/>)

    expect(figure.exists('div')).toBe(true)
    expect(figure.find('div').prop('style')).toHaveProperty('width', expectedWidth)
  })
})
