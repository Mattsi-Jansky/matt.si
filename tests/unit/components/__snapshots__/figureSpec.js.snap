// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Figure hasn't changed 1`] = `
<div
  id="content-figure"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <figure>
    <a
      href="/static/45b26cc583de29f73117dce640a86d75/8a760/banner.jpg"
      target="_blank"
    >
      <div
        className=" gatsby-image-wrapper"
        style={
          Object {
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "paddingBottom": "54.69301340860974%",
              "width": "100%",
            }
          }
        />
        <img
          alt=""
          src="data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAALABQDASIAAhEBAxEB/8QAFwABAQEBAAAAAAAAAAAAAAAAAwABBP/EABUBAQEAAAAAAAAAAAAAAAAAAAAB/9oADAMBAAIQAxAAAAES6MFko//EABoQAAICAwAAAAAAAAAAAAAAAAECAAMREyH/2gAIAQEAAQUCNx17OclKKy4AmJ//xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAEDAQE/AT//xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAECAQE/AT//xAAeEAACAQMFAAAAAAAAAAAAAAAAARICIYEQEUFRYf/aAAgBAQAGPwKLqt6Rkng5wi6Kdu9P/8QAHRAAAgEEAwAAAAAAAAAAAAAAAAERITFBUWGBkf/aAAgBAQABPyFm4M1E7Sb2Oc+wamtug3YokOW/T//aAAwDAQACAAMAAAAQEz//xAAXEQADAQAAAAAAAAAAAAAAAAAAAREh/9oACAEDAQE/EGrpEf/EABYRAQEBAAAAAAAAAAAAAAAAAAABEf/aAAgBAgEBPxCVr//EAB8QAQACAQMFAAAAAAAAAAAAAAEAESExQVFhcZHR8f/aAAgBAQABPxBeEekldrM/ICQMFAPF2wuXceUeSFWGDpt0imNAa3ISy441+5//2Q=="
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 1,
              "position": "absolute",
              "top": 0,
              "transitionDelay": "500ms",
              "width": "100%",
            }
          }
        />
        <picture>
          <source
            sizes="(max-width: 1240px) 100vw, 1240px"
            srcSet="/static/45b26cc583de29f73117dce640a86d75/68709/banner.jpg 310w,
/static/45b26cc583de29f73117dce640a86d75/53593/banner.jpg 620w,
/static/45b26cc583de29f73117dce640a86d75/8a760/banner.jpg 1240w,
/static/45b26cc583de29f73117dce640a86d75/0e6ff/banner.jpg 1860w,
/static/45b26cc583de29f73117dce640a86d75/7efdd/banner.jpg 2480w,
/static/45b26cc583de29f73117dce640a86d75/f2bea/banner.jpg 2834w"
          />
          <img
            alt=""
            loading="lazy"
            onLoad={[Function]}
            sizes="(max-width: 1240px) 100vw, 1240px"
            src="/static/45b26cc583de29f73117dce640a86d75/8a760/banner.jpg"
            srcSet="/static/45b26cc583de29f73117dce640a86d75/68709/banner.jpg 310w,
/static/45b26cc583de29f73117dce640a86d75/53593/banner.jpg 620w,
/static/45b26cc583de29f73117dce640a86d75/8a760/banner.jpg 1240w,
/static/45b26cc583de29f73117dce640a86d75/0e6ff/banner.jpg 1860w,
/static/45b26cc583de29f73117dce640a86d75/7efdd/banner.jpg 2480w,
/static/45b26cc583de29f73117dce640a86d75/f2bea/banner.jpg 2834w"
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 500ms",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><source srcset=\\"/static/45b26cc583de29f73117dce640a86d75/68709/banner.jpg 310w,
/static/45b26cc583de29f73117dce640a86d75/53593/banner.jpg 620w,
/static/45b26cc583de29f73117dce640a86d75/8a760/banner.jpg 1240w,
/static/45b26cc583de29f73117dce640a86d75/0e6ff/banner.jpg 1860w,
/static/45b26cc583de29f73117dce640a86d75/7efdd/banner.jpg 2480w,
/static/45b26cc583de29f73117dce640a86d75/f2bea/banner.jpg 2834w\\" sizes=\\"(max-width: 1240px) 100vw, 1240px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 1240px) 100vw, 1240px\\" srcset=\\"/static/45b26cc583de29f73117dce640a86d75/68709/banner.jpg 310w,
/static/45b26cc583de29f73117dce640a86d75/53593/banner.jpg 620w,
/static/45b26cc583de29f73117dce640a86d75/8a760/banner.jpg 1240w,
/static/45b26cc583de29f73117dce640a86d75/0e6ff/banner.jpg 1860w,
/static/45b26cc583de29f73117dce640a86d75/7efdd/banner.jpg 2480w,
/static/45b26cc583de29f73117dce640a86d75/f2bea/banner.jpg 2834w\\" src=\\"/static/45b26cc583de29f73117dce640a86d75/8a760/banner.jpg\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </a>
    <figcaption>
      Image from 
      <a
        href="foo.bar"
      >
        Foo Bar
      </a>
    </figcaption>
  </figure>
</div>
`;
